### PR TITLE
Move `os_str_bytes` to `sys::unix`

### PR DESCRIPTION
**Pull request by CDirkx**
Followup to #84967, with `OsStrExt` and `OsStringExt` moved out of `sys_common`, there is no reason anymore for `os_str_bytes` to live in `sys_common` and not in sys. This pr moves it to the location `sys::unix::os_str` and reuses the code on other platforms via `#[path]` (as is common in `sys`) instead of importing.

### CODE REVIEW

File: library/std/src/sys_common/mod.rs [hunk 1]
```diff
@@ -27,10 +27,6 @@ pub mod fs;
 pub mod io;
 pub mod memchr;
 pub mod mutex;
-// `doc` is required because `sys/mod.rs` imports `unix/ext/mod.rs` on Windows
-// when generating documentation.
-#[cfg(any(doc, not(windows)))]
```
Comment (CDirkx) [inline library/std/src/sys_common/mod.rs:32]: Not needed anymore, this now completely falls under the logic in `std::os`

File: library/std/src/sys/unsupported/common.rs [hunk 2]
```diff
@@ -4,8 +4,6 @@ pub mod memchr {
     pub use core::slice::memchr::{memchr, memrchr};
 }
 
-pub use crate::sys_common::os_str_bytes as os_str;
```
Comment (CDirkx) [inline library/std/src/sys/unsupported/common.rs:7]: Should `os_str` be here? Both `wasi` and `wasm` do `use common::*` but define `os_str` themselves anyway.

File: library/std/src/sys/unix/os_str.rs [hunk 3]
```diff
@@ -13,6 +13,7 @@ use crate::sys_common::{AsInner, IntoInner};
 use core::str::lossy::{Utf8Lossy, Utf8LossyChunk};
 
 #[cfg(test)]
+#[path = "../unix/os_str/tests.rs"]
```
Comment (CDirkx) [inline library/std/src/sys/unix/os_str.rs:16]: To avoid duplicating the test, could alternatively be `cfg(unix)`.

**Comment by m-ou-se** [comment]
Thanks!

@bors r+

**Pull request merged**